src/OpenRiaServices.Server/Framework/Data/DomainOperationParameter.cs

```diff
@@ -107,5 +110,50 @@ public override string ToString()
         {
             return string.Format(CultureInfo.InvariantCulture, "{0} {1}", this.ParameterType, this.Name);
         }
+
+        private Nullable<bool> _HasValidationAttribute;
+        /// <summary>
+        /// Returns true if this type has validation attribute
+        /// </summary>
+        public bool HasValidationAttributeOnProperties
+        {
+            get
+            {
+                if (!_HasValidationAttribute.HasValue)
+                {
+                    if (TypeUtility.IsPredefinedSimpleType(this._parameterType))
+                        _HasValidationAttribute = false;
+                    else
+                    {
+                        foreach (var Prop in _parameterType.GetProperties())
```

Comment:
See my comment in CalculateRequiresValidation I think we should be able to use the MetaType for all types which are note primitive